control-plane-operator/hostedclusterconfigoperator/controllers/nodecount/setup.go

```diff
@@ -0,0 +1,40 @@
+package nodecount
+
+import (
+	"fmt"
+
+	hypershiftclient "github.com/openshift/hypershift/client/clientset/clientset"
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/operator"
+)
+
+const ControllerName = "nodecount"
+
+func Setup(opts *operator.HostedClusterConfigOperatorConfig) error {
+	hypershiftClient, err := hypershiftclient.NewForConfig(opts.CPCluster.GetConfig())
+	if err != nil {
+		return err
+	}
+
+	r := &reconciler{
+		hcpName:            opts.HCPName,
+		hcpNamespace:       opts.Namespace,
+		client:             hypershiftClient,
+		guestClusterClient: opts.Manager.GetClient(),
+	}
+	c, err := controller.New(ControllerName, opts.Manager, controller.Options{Reconciler: r})
```

Comment:
controllers in this operator predate the new builder syntax for controller-runtime, but since this is a new controller we should use it as in https://github.com/openshift/hypershift/blob/d445171917b35b43a6dceba4e676871f297687b8/control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go#L182-L192